### DISCUSSION TITLE
Zigbee2MQTT suddenly stops after 2 weeks normal run (not connected to MQTT server)

### DISCUSSION THREAD

**Discussion by McGyver64** [Koenkk/zigbee2mqtt#6899] [Devices]
Running Domoticz on Raspberry Pi4 with cc2652rb_stick_-_slaesh Stick on TTYUSB0

I had this problem for 3 times now, first I thought it was the CC2531 stick and upgraded to Sleash stick, but this is second time in 2 months the Pi "died" and zigbee2mqtt stops functioning. 
2 weeks ago complete fresh install, updated everything etc.
Fresh install of Domoticz and Zigbee2MQTT and adding several devices. Also using some Hue device in Domiticz, there is where I find there is something wrong. All Hue devices are being read and updated by Domoticz, but all zigbee2mqtt devices had last reading on 23:30 yesterday. Also Zigbee2mqtt does not show devices/groups/Network Map in Domoticz
I was testing/running some Blockly Event scripts yesterday at 22:00-00:30 and found result little strange, so stopped and went to bed. Only this morning noticing that zigbee2mqtt is not updating/reading devices as of 22:53 - 23:30 (last readings) 

logged in with ssh and did some test

Input = ls -l /dev/serial/by-id
output = sb-Silicon_Labs_slae.sh_cc2652rb_stick_-_slaesh_s_iot_stuff_00_12_4B_00_21_CC_37_E9-if00-port0 -> ../../ttyUSB0
version = mosquitto version 1.5.7
Input = mosquitto_pub -h localhost -t zigbee2mqtt -m "hello world"
output = Error: Connection refused

npm start gives error "not connected to mqtt server" 

>npm start

> zigbee2mqtt@1.18.1 start /opt/zigbee2mqtt
> node index.js

Zigbee2MQTT:info  2021-03-31 12:10:52: Logging to console and directory: '/opt/zigbee2mqtt/data/log/2021-03-31.12-10-51' filename: log.txt
Zigbee2MQTT:info  2021-03-31 12:10:52: Starting Zigbee2MQTT version 1.18.1 (commit #a43d1f3)
Zigbee2MQTT:info  2021-03-31 12:10:52: Starting zigbee-herdsman (0.13.71)
Zigbee2MQTT:info  2021-03-31 12:10:55: zigbee-herdsman started
Zigbee2MQTT:info  2021-03-31 12:10:55: Coordinator firmware version: '{"meta":{"maintrel":1,"majorrel":2,"minorrel":7,"product":1,"revision":20200715,"transportrev":2},"type":"zStack3x0"}'
Zigbee2MQTT:info  2021-03-31 12:10:55: Currently 20 devices are joined:
Zigbee2MQTT:info  2021-03-31 12:10:55: 0x588e81fffed39f9b (0x588e81fffed39f9b): TS0121_plug - TuYa 10A UK or 16A EU smart plug (Router)
Zigbee2MQTT:info  2021-03-31 12:10:55: 0x00158d0004a008fb (0x00158d0004a008fb): WSDCGQ11LM - Xiaomi Aqara temperature, humidity and pressure sensor (EndDevice)
Zigbee2MQTT:info  2021-03-31 12:10:55: 0x842e14fffe626176 (0x842e14fffe626176): HG06337 - Lidl Silvercrest smart plug (EU, CH, FR, BS, DK) (Router)
Zigbee2MQTT:info  2021-03-31 12:10:55: Lidl Motion (0xbc33acfffe5e4ee2): HG06335 - Lidl Silvercrest smart motion sensor (EndDevice)
Zigbee2MQTT:info  2021-03-31 12:10:55: 0x00158d000485254c (0x00158d000485254c): WSDCGQ11LM - Xiaomi Aqara temperature, humidity and pressure sensor (EndDevice)
Zigbee2MQTT:info  2021-03-31 12:10:55: 0x5c0272fffec9da73 (0x5c0272fffec9da73): TS0601_thermostat - TuYa Radiator valve with thermostat (EndDevice)
Zigbee2MQTT:info  2021-03-31 12:10:55: 0x5c0272fffedfbca0 (0x5c0272fffedfbca0): TS0601_thermostat - TuYa Radiator valve with thermostat (EndDevice)
Zigbee2MQTT:info  2021-03-31 12:10:55: 0xbc33acfffe458434 (0xbc33acfffe458434): TS0601_thermostat - TuYa Radiator valve with thermostat (EndDevice)
Zigbee2MQTT:info  2021-03-31 12:10:55: 0x60a423fffeab24a7 (0x60a423fffeab24a7): TS0601_thermostat - TuYa Radiator valve with thermostat (EndDevice)
Zigbee2MQTT:info  2021-03-31 12:10:55: 0x00158d000485259f (0x00158d000485259f): WSDCGQ11LM - Xiaomi Aqara temperature, humidity and pressure sensor (EndDevice)
Zigbee2MQTT:info  2021-03-31 12:10:55: 0x842e14fffe3666ef (0x842e14fffe3666ef): TS0121_plug - TuYa 10A UK or 16A EU smart plug (Router)
Zigbee2MQTT:info  2021-03-31 12:10:55: 0x588e81fffed393ad (0x588e81fffed393ad): TS0121_plug - TuYa 10A UK or 16A EU smart plug (Router)
Zigbee2MQTT:info  2021-03-31 12:10:55: 0x00158d00054b6c1c (0x00158d00054b6c1c): WSDCGQ11LM - Xiaomi Aqara temperature, humidity and pressure sensor (EndDevice)
Zigbee2MQTT:info  2021-03-31 12:10:55: 0x00158d00054be2ad (0x00158d00054be2ad): WSDCGQ11LM - Xiaomi Aqara temperature, humidity and pressure sensor (EndDevice)
Zigbee2MQTT:info  2021-03-31 12:10:55: 0x00158d00048525dc (0x00158d00048525dc): WSDCGQ11LM - Xiaomi Aqara temperature, humidity and pressure sensor (EndDevice)
Zigbee2MQTT:info  2021-03-31 12:10:55: 0x00158d00054bdd02 (0x00158d00054bdd02): WSDCGQ11LM - Xiaomi Aqara temperature, humidity and pressure sensor (EndDevice)
Zigbee2MQTT:info  2021-03-31 12:10:55: 0x00158d00054be26f (0x00158d00054be26f): WSDCGQ11LM - Xiaomi Aqara temperature, humidity and pressure sensor (EndDevice)
Zigbee2MQTT:info  2021-03-31 12:10:55: 0x5c0272fffedab20a (0x5c0272fffedab20a): TS0601_thermostat - TuYa Radiator valve with thermostat (EndDevice)
Zigbee2MQTT:info  2021-03-31 12:10:55: 0x847127fffe0eba58 (0x847127fffe0eba58): TS0601_thermostat - TuYa Radiator valve with thermostat (EndDevice)
Zigbee2MQTT:info  2021-03-31 12:10:55: 0x588e81fffeff5cbb (0x588e81fffeff5cbb): TS0121_plug - TuYa 10A UK or 16A EU smart plug (Router)
Zigbee2MQTT:warn  2021-03-31 12:10:55: `permit_join` set to  `true` in configuration.yaml.
Zigbee2MQTT:warn  2021-03-31 12:10:55: Allowing new devices to join.
Zigbee2MQTT:warn  2021-03-31 12:10:55: Set `permit_join` to `false` once you joined all devices.
Zigbee2MQTT:info  2021-03-31 12:10:55: Zigbee: allowing new devices to join.
Zigbee2MQTT:info  2021-03-31 12:10:55: Connecting to MQTT server at mqtt://localhost
Zigbee2MQTT:error 2021-03-31 12:10:56: Not connected to MQTT server!
Zigbee2MQTT:error 2021-03-31 12:13:21: Cannot send message: topic: 'zigbee2mqtt/bridge/state', payload: 'offline
Zigbee2MQTT:info  2021-03-31 12:13:21: Disconnecting from MQTT server
Zigbee2MQTT:info  2021-03-31 12:13:21: zigbee-herdsman stopped
Zigbee2MQTT:info  2021-03-31 12:13:21: Stopped Zigbee2MQTT

After a reboot of the Pi still no respons 

But when running "mosquitto -v" zigbee2mqtt is starting reading devices and updating Domoticz.  devices/groups/Network Map also shows again. But as soon as ctrl-c is entered,  npm gives error, devices/groups/Network Map are not showed anymore.

mosquitto -v
1617186492: mosquitto version 1.5.7 starting
1617186492: Using default config.
1617186492: Opening ipv4 listen socket on port 1883.
1617186492: Opening ipv6 listen socket on port 1883.
1617186492: New connection from 127.0.0.1 on port 1883.
1617186492: New client connected from 127.0.0.1 as mqttjs_7260d58d (c1, k60).
1617186492: Will message specified (7 bytes) (r1, q0).
1617186492: 	zigbee2mqtt/bridge/state
1617186492: Sending CONNACK to mqttjs_7260d58d (0, 0)
1617186492: Received SUBSCRIBE from mqttjs_7260d58d
1617186492: 	zigbee2mqtt/# (QoS 0)
1617186492: mqttjs_7260d58d 0 zigbee2mqtt/#
1617186492: Sending SUBACK to mqttjs_7260d58d
1617186492: Received PUBLISH from mqttjs_7260d58d (d0, q0, r0, m0, 'zigbee2mqtt/bridge/logging', ... (53 bytes))
1617186492: Sending PUBLISH to mqttjs_7260d58d (d0, q0, r0, m0, 'zigbee2mqtt/bridge/logging', ... (53 bytes))
1617186492: Received PUBLISH from mqttjs_7260d58d (d0, q0, r0, m0, 'zigbee2mqtt/bridge/logging', ... (93 bytes))
1617186492: Sending PUBLISH to mqttjs_7260d58d (d0, q0, r0, m0, 'zigbee2mqtt/bridge/logging', ... (93 bytes))
1617186492: Received PUBLISH from mqttjs_7260d58d (d0, q0, r1, m0, 'zigbee2mqtt/bridge/state', ... (6 bytes))
1617186492: Sending PUBLISH to mqttjs_7260d58d (d0, q0, r0, m0, 'zigbee2mqtt/bridge/state', ... (6 bytes))
1617186495: Received PUBLISH from mqttjs_7260d58d (d0, q0, r0, m0, 'zigbee2mqtt/bridge/logging', ... (1313 bytes))
1617186495: Sending PUBLISH to mqttjs_7260d58d (d0, q0, r0, m0, 'zigbee2mqtt/bridge/logging', ... (1313 bytes))
1617186495: Received PUBLISH from mqttjs_7260d58d (d0, q0, r0, m0, 'zigbee2mqtt/0x5c0272fffedfbca0', ... (1072 bytes))
1617186495: Sending PUBLISH to mqttjs_7260d58d (d0, q0, r0, m0, 'zigbee2mqtt/0x5c0272fffedfbca0', ... (1072 bytes))
1617186495: Received PUBLISH from mqttjs_7260d58d (d0, q0, r0, m0, 'zigbee2mqtt/bridge/logging', ... (1312 bytes))
1617186495: Sending PUBLISH to mqttjs_7260d58d (d0, q0, r0, m0, 'zigbee2mqtt/bridge/logging', ... (1312 bytes))
1617186495: Received PUBLISH from mqttjs_7260d58d (d0, q0, r0, m0, 'zigbee2mqtt/0x5c0272fffedfbca0', ... (1071 bytes))
1617186495: Sending PUBLISH to mqttjs_7260d58d (d0, q0, r0, m0, 'zigbee2mqtt/0x5c0272fffedfbca0', ... (1071 bytes))
1617186496: New connection from 127.0.0.1 on port 1883.
1617186496: New client connected from 127.0.0.1 as Domoticz_1617186033_9976 (c1, k60).
1617186496: No will message specified.

-ended with "ctrl-c" 

npm start
npm ERR! code ENOENT
npm ERR! syscall open
npm ERR! path /home/pi/package.json
npm ERR! errno -2
npm ERR! enoent ENOENT: no such file or directory, open '/home/pi/package.json'
npm ERR! enoent This is related to npm not being able to find a file.
npm ERR! enoent 

npm ERR! A complete log of this run can be found in:
npm ERR!     /home/pi/.npm/_logs/2021-03-31T10_25_54_145Z-debug.log

**Comment by Koenkk** [comment]
`Coordinator firmware version: '{"meta":{"maintrel":1,"majorrel":2,"minorrel":7,"product":1,"revision":20200715,"transportrev":2},"type":"zStack3x0"}'` you are using an old firmware version, try updating to the latest which has a fix for crashing after some uptime.

**Reply by McGyver64** [reply]
you mean a update of the cc2652rb USB stick? (bought it only 1 month ago) or Zigbee2MQTT software? 
Strange thing is that this is happening for 5-6 th time now, first 3 times with a CC2531 stick, wich, I thought was the reason for crashing. But now with CC2652 stick the same experience.

**Reply by McGyver64** [reply]
I started 1 year ago with Z2M on a Pi Zero W with a CC2531 device. After several weeks the Pi crashed. I thought it was my "newbie" status. Fixing, reinstalling Z2M, broker, Domoticz was not working, so I did a new/fresh install, including pairing all devices. But after 1 month the same shit happened, so again new install because repair did not work. Luckily I made a Domoticz backup, but again ail devices have to be paired again. 
After the 3th crash I thought it was the Pi Zero W to be the bad guy, so replaced it with a Pi 4. Again new install and all seemed fine, but after some weeks, crash again. This time I thought the CC2531 was the reason for crash, because there were 20+ devices and respons was really slow sometimes. So I bought the CC2653RB stick arrived early jan 2021. again new install of every thing, pairing devices etc and up and running......... for 3-4 weeks and crash again. Tried repair, update, reinstalling broker etc. Mosquitto -v made communication come back again, but after one hour there were no more updates of devices and showing above logs (topic start) . With 2 different Pi and 2 different Z2M devices I get similar situation. The only "solid" factor is Nefit Easy over http, Domoticz, Z2M Broker, NodeJS, NPM and Zigbee devices (tuya temp sensor, radiator thermostat, Blitzwolf 220 outlet) .

The reason for my Pi Situation is to regulate upstate temperature while the CV thermostat is in living room and when temp = 20C , upstairs is not heated anymore. By using Zigbee Tuya Thermostat I can close radiatorsdownstares and open upstairs ,and with Temp Sensors I can manipulate/regulate the Nefit thermostat and create a situation where up and downstairs the temperature is regulated .  My "feeling" tells me that it is going wrong when I program the "heating / regulating part" and play with some settings. https://prnt.sc/10zuqwy
Because Hue is also connected (lights outlets, etc) and this keeps running, it does not come to my direct attention that Z2M is stopped at a certain time. 

Today I got a reply from supplier of CC2653RB, wich suggested it could be the Z2M broker. It is the same broker in all the  installations ??

**Reply by Koenkk** [reply]
I mean the firmware version. You are running an old version which has a known bug about this: https://github.com/Koenkk/zigbee2mqtt/discussions/6019

**Comment by McGyver64** [comment]
I tried to flash the stick with the latest firmware and followed this link. 

https://slae.sh/projects/cc2652/    I updated python to 3.8.0 like suggested

Before running I stopped the services Domiticz, Zigbee2mqtt and Mosquitto 
But I get an error while flashing. I also get this error when I use the "serial by ID" naming. 
cc2538 is the directory created by boot loader tool

pi@Domoticz:~/cc2538-bsl $ ./cc2538-bsl.py -p USB0 -evw CC2652RB_20210120.hex
ERROR: [Errno 2] could not open port USB0: [Errno 2] No such file or directory: 'USB0'

I also tried the Uniflsh tool from TI version 6.2.x (windows) both this software does not detect the Stick when it is inserted in my PC (MacBook Air with bootcamp, with or without holding the BL button during insert)

help and/or assist is very welcome :-)

**Reply by lux73** [reply]
try /dev/ttyUSB0 - USB0 is not enough

**Reply by McGyver64** [reply]
I was able to do the flash on an Intel NUC 6 with Windows 10, using the TI Uni Flash tool. This time the USB device was detected and the tool has successfully be flashing the USB device.

Domoticz is now showing v.1.18.2 (zStack3x0 20210120), wich should be the correct/latest version
